src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ParameterAssignmentCheck.java

```diff
@@ -143,90 +156,36 @@ public void beginTree(DetailAST rootAST) {
 
     @Override
     public void visitToken(DetailAST ast) {
-        switch (ast.getType()) {
-            case TokenTypes.CTOR_DEF:
-            case TokenTypes.METHOD_DEF:
-                visitMethodDef(ast);
-                break;
-            case TokenTypes.ASSIGN:
-            case TokenTypes.PLUS_ASSIGN:
-            case TokenTypes.MINUS_ASSIGN:
-            case TokenTypes.STAR_ASSIGN:
-            case TokenTypes.DIV_ASSIGN:
-            case TokenTypes.MOD_ASSIGN:
-            case TokenTypes.SR_ASSIGN:
-            case TokenTypes.BSR_ASSIGN:
-            case TokenTypes.SL_ASSIGN:
-            case TokenTypes.BAND_ASSIGN:
-            case TokenTypes.BXOR_ASSIGN:
-            case TokenTypes.BOR_ASSIGN:
-                visitAssign(ast);
-                break;
-            case TokenTypes.INC:
-            case TokenTypes.POST_INC:
-            case TokenTypes.DEC:
-            case TokenTypes.POST_DEC:
-                visitIncDec(ast);
-                break;
-            default:
-                throw new IllegalStateException(ast.toString());
+        final int type = ast.getType();
+        if (TokenUtil.isOfType(type, TokenTypes.CTOR_DEF, TokenTypes.METHOD_DEF)) {
+            visitMethodDef(ast);
+        }
+        else if (type == TokenTypes.LAMBDA) {
+            if (ast.getParent().getType() != TokenTypes.SWITCH_RULE) {
+                visitLambda(ast);
+            }
+        }
+        else {
+            checkNestedIdent(ast);
```

Comment:
I had to do this due to pitest but I personally don't like this change. What this does is call the method `checkIdent(ast)` for every token other than `[TokenTypes.CTOR_DEF,TokenTypes.METHOD_DEF,TokenTypes.LAMBDA]`
logically it is correct but not very maintainable, in the future if we would have to add another token, it can take a while to understand what's going on, I renamed it to help make some more sense.